akka-remote/src/test/scala/akka/remote/artery/TlsTcpSpec.scala

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import scala.concurrent.duration._
+
+import akka.actor.ActorIdentity
+import akka.actor.Identify
+import akka.actor.RootActorPath
+import akka.testkit.ImplicitSender
+import akka.testkit.TestActors
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+
+class TlsTcpWithDefaultConfigSpec extends TlsTcpSpec(ConfigFactory.parseString("""
+    akka.remote.artery {
+      transport = tls-tcp
+    }
+    """).withFallback(ArterySpecSupport.defaultConfig))
+
+// FIXME test more config combinations, see Ticket1978CommunicationSpec
+
+object TlsTcpSpec {
+
+  def keystoreConfig(): Config = {
+    val trustStore = getClass.getClassLoader.getResource("truststore").getPath
+    val keyStore = getClass.getClassLoader.getResource("keystore").getPath
+
+    ConfigFactory.parseString(s"""
+      akka.remote.artery.ssl {
+          key-store = "$keyStore"
+          trust-store = "$trustStore"
+      }
+    """)
+  }
+
+}
+
+abstract class TlsTcpSpec(config: Config)
+  extends ArteryMultiNodeSpec(config.withFallback(TlsTcpSpec.keystoreConfig())) with ImplicitSender {
+
+  val systemB = newRemoteSystem(name = Some("systemB"))
+  val addressB = address(systemB)
+  val rootB = RootActorPath(addressB)
+
+  "Artery with TLS/TCP" must {
+
+    "deliver messages" in {
+      systemB.actorOf(TestActors.echoActorProps, "echo")
+
+      val echoRef = {
+        system.actorSelection(rootB / "user" / "echo") ! Identify(None)
+        expectMsgType[ActorIdentity](5.seconds).ref.get
+      }
+
+      echoRef ! "ping-1"
+      expectMsg("ping-1")
+
+      // and some more
+      (2 to 10).foreach { n ⇒
+        echoRef ! s"ping-$n"
+      }
+      receiveN(9) should equal((2 to 10).map(n ⇒ s"ping-$n"))
+    }
+
+    "deliver messages over large messages stream" in {
+      // FIXME
+      pending
+    }
+  }
+
+}
```

Comment:
Does anyone have any hint of why it can't stop the AS?

```
Failed to stop [TlsTcpWithBasicConfigSpec] within [10 seconds]
-> / LocalActorRefProvider$$anon$1 class akka.actor.LocalActorRefProvider$Guardian status=0 1 children
   ⌊-> system LocalActorRef class akka.actor.LocalActorRefProvider$SystemGuardian status=0 11 children
       ⌊-> IO-DNS RepointableActorRef class akka.io.SimpleDnsManager status=0 1 children
       |   ⌊-> inet-address RoutedActorRef class akka.routing.RouterPoolActor status=0 4 children
       |       ⌊-> $a LocalActorRef class akka.io.InetAddressDnsResolver status=0 no children
       |       ⌊-> $b LocalActorRef class akka.io.InetAddressDnsResolver status=0 no children
       |       ⌊-> $c LocalActorRef class akka.io.InetAddressDnsResolver status=0 no children
       |       ⌊-> $d LocalActorRef class akka.io.InetAddressDnsResolver status=0 no children
       ⌊-> IO-TCP RepointableActorRef class akka.io.TcpManager status=0 1 children
       |   ⌊-> selectors RoutedActorRef class akka.routing.RouterPoolActor status=0 1 children
       |       ⌊-> $a LocalActorRef class akka.io.SelectionHandler status=0 5 children
       |           ⌊-> 0 LocalActorRef class akka.io.TcpListener status=0 no children
       |           ⌊-> 1 LocalActorRef class akka.io.TcpOutgoingConnection status=0 no children
       |           ⌊-> 2 LocalActorRef class akka.io.TcpOutgoingConnection status=0 no children
       |           ⌊-> 3 LocalActorRef class akka.io.TcpIncomingConnection status=0 no children
       |           ⌊-> 4 LocalActorRef class akka.io.TcpIncomingConnection status=0 no children
       ⌊-> StreamSupervisor-0 RepointableActorRef class akka.stream.impl.StreamSupervisor status=0 9 children
       |   ⌊-> remote-10-1 RepointableActorRef class akka.stream.impl.io.TLSActor status=0 no children
       |   ⌊-> remote-11-0-map RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-11-1 RepointableActorRef class akka.stream.impl.io.TLSActor status=0 no children
       |   ⌊-> remote-12-0-map RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-12-1 RepointableActorRef class akka.stream.impl.io.TLSActor status=0 no children
       |   ⌊-> remote-6-0-ignoreSink RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-7-0-ignoreSink RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-8-0-ignoreSink RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-9-1 RepointableActorRef class akka.stream.impl.io.TLSActor status=0 no children
       ⌊-> StreamSupervisor-1 RepointableActorRef class akka.stream.impl.StreamSupervisor status=0 no children
       ⌊-> deadLetterListener RepointableActorRef class akka.event.DeadLetterListener status=0 no children
       ⌊-> eventStreamUnsubscriber-1 RepointableActorRef class akka.event.EventStreamUnsubscriber status=0 no children
       ⌊-> log1-TestEventListener RepointableActorRef class akka.testkit.TestEventListener status=0 no children
       ⌊-> remote-deployment-watcher RepointableActorRef class akka.remote.RemoteDeploymentWatcher status=0 no children
       ⌊-> remote-watcher RepointableActorRef class akka.remote.RemoteWatcher status=0 no children
       ⌊-> remoting-terminator RepointableActorRef class akka.remote.RemoteActorRefProvider$RemotingTerminator status=0 no children
       ⌊-> testActor-1 RepointableActorRef class akka.testkit.TestActor status=0 no children
Failed to stop [systemB] within [10 seconds]
-> / LocalActorRefProvider$$anon$1 class akka.actor.LocalActorRefProvider$Guardian status=0 1 children
   ⌊-> system LocalActorRef class akka.actor.LocalActorRefProvider$SystemGuardian status=0 10 children
       ⌊-> IO-DNS RepointableActorRef class akka.io.SimpleDnsManager status=0 1 children
       |   ⌊-> inet-address RoutedActorRef class akka.routing.RouterPoolActor status=0 4 children
       |       ⌊-> $a LocalActorRef class akka.io.InetAddressDnsResolver status=0 no children
       |       ⌊-> $b LocalActorRef class akka.io.InetAddressDnsResolver status=0 no children
       |       ⌊-> $c LocalActorRef class akka.io.InetAddressDnsResolver status=0 no children
       |       ⌊-> $d LocalActorRef class akka.io.InetAddressDnsResolver status=0 no children
       ⌊-> IO-TCP RepointableActorRef class akka.io.TcpManager status=0 1 children
       |   ⌊-> selectors RoutedActorRef class akka.routing.RouterPoolActor status=0 1 children
       |       ⌊-> $a LocalActorRef class akka.io.SelectionHandler status=0 5 children
       |           ⌊-> 0 LocalActorRef class akka.io.TcpListener status=0 no children
       |           ⌊-> 1 LocalActorRef class akka.io.TcpIncomingConnection status=0 no children
       |           ⌊-> 2 LocalActorRef class akka.io.TcpIncomingConnection status=0 no children
       |           ⌊-> 3 LocalActorRef class akka.io.TcpOutgoingConnection status=0 no children
       |           ⌊-> 4 LocalActorRef class akka.io.TcpOutgoingConnection status=0 no children
       ⌊-> StreamSupervisor-2 RepointableActorRef class akka.stream.impl.StreamSupervisor status=0 9 children
       |   ⌊-> remote-10-0-ignoreSink RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-11-1 RepointableActorRef class akka.stream.impl.io.TLSActor status=0 no children
       |   ⌊-> remote-12-1 RepointableActorRef class akka.stream.impl.io.TLSActor status=0 no children
       |   ⌊-> remote-6-0-ignoreSink RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-7-0-map RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-7-1 RepointableActorRef class akka.stream.impl.io.TLSActor status=0 no children
       |   ⌊-> remote-8-0-map RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       |   ⌊-> remote-8-1 RepointableActorRef class akka.stream.impl.io.TLSActor status=0 no children
       |   ⌊-> remote-9-0-ignoreSink RepointableActorRef class akka.stream.impl.fusing.ActorGraphInterpreter status=0 no children
       ⌊-> StreamSupervisor-3 RepointableActorRef class akka.stream.impl.StreamSupervisor status=0 no children
       ⌊-> deadLetterListener RepointableActorRef class akka.event.DeadLetterListener status=0 no children
       ⌊-> eventStreamUnsubscriber-2 RepointableActorRef class akka.event.EventStreamUnsubscriber status=0 no children
       ⌊-> log1-TestEventListener RepointableActorRef class akka.testkit.TestEventListener status=0 no children
       ⌊-> remote-deployment-watcher RepointableActorRef class akka.remote.RemoteDeploymentWatcher status=0 no children
       ⌊-> remote-watcher RepointableActorRef class akka.remote.RemoteWatcher status=0 no children
       ⌊-> remoting-terminator RepointableActorRef class akka.remote.RemoteActorRefProvider$RemotingTerminator status=0 no children
```

Reply:
I have fixed this.